library/psa_crypto.c

```diff
@@ -3453,6 +3506,8 @@ static psa_status_t psa_rsa_sign( mbedtls_rsa_context *rsa,
     psa_status_t status;
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
     mbedtls_md_type_t md_alg;
+    (void)hash;
```

Comment:
Those two parameters are used in both signature algorithms and the function is guarded by
```c
#if defined(MBEDTLS_PSA_BUILTIN_ALG_RSA_PKCS1V15_SIGN) || \
    defined(MBEDTLS_PSA_BUILTIN_ALG_RSA_PSS)
```
thus I think you can just remove those two line.